### PR TITLE
doc: update links in ONBOARDING to new branches

### PR DESCRIPTION
**Pull request by tniessen**
Refs: https://github.com/nodejs/node/issues/33864

### CODE REVIEW

File: ONBOARDING.md [hunk 1]
```diff
@@ -1,15 +1,15 @@
 # Onboarding/Offboarding
 
 Please refer to the
-[Establishment of the TSC](https://github.com/nodejs/TSC/blob/master/TSC-Charter.md#section-4-establishment-of-the-tsc)
+[Establishment of the TSC](https://github.com/nodejs/TSC/blob/main/TSC-Charter.md#section-4-establishment-of-the-tsc)
```
[the remark this answers is not in the document]
Comment (tniessen) [reply ONBOARDING.md:4]: demonstration of the issue with HEAD

[the remark this answers is not in the document]
Comment (ljharb) [reply ONBOARDING.md:4]: yeah, that is true. how often are people expected to edit the charter?

[the remark this answers is not in the document]
Comment (tniessen) [reply ONBOARDING.md:4]: Some of the links point to files that are supposed to be modified as part of the documented onboarding/offboarding processes. Either way, I'd expect files to be edited more frequently than we are going to rename the `main` branch.

I don't care much either way, if you think `HEAD` is better for some reason, I'm happy to change it.

[the remark this answers is not in the document]
Comment (tniessen) [reply ONBOARDING.md:4]: > it just seems nice to avoid hardcoding

💯 agreed

**Review by RaisinTen — APPROVED** [review]

**Pull request merged**